### DISCUSSION TITLE
Trying to setup ingress resulting in back-off-restarting with 'error retrieving k8s version'

### DISCUSSION THREAD

**Discussion by typoworx-de** [nginx/kubernetes-ingress#2664] [Q&A]
checked out this git-repo and tried to deploy on fresh-setup kubernetes (bare-metal, 2 workers) with flannel-cni.

I used these commands to setup:

kubectl apply -f common/ns-and-sa.yaml
kubectl apply -f common/default-server-secret.yaml
kubectl apply -f common/nginx-config.yaml
kubectl apply -f rbac/rbac.yaml
kubectl apply -f deployment/nginx-ingress.yaml
kubectl apply -f service/loadbalancer.yaml

After some time everything started up, except "nginx-ingress":
State is 'waiting'
Last Status: 'Terminate' with Reason 'exit-code 255'

Back-off restarting failed container
```
Readiness probe failed: Get "http://10.96.2.4:8081/nginx-ready": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Terminal Log:
```
I0506 13:41:57.751680       1 main.go:213] Starting NGINX Ingress Controller Version=2.2.0 GitCommit=72602e19f0bb160cfe0cfa791edd922e3842def2 Date=2022-04-12T22:46:02Z Arch=linux/amd64 PlusFlag=false
F0506 13:42:27.754322       1 main.go:342] error retrieving k8s version: Get "https://10.10.0.1:443/version": dial tcp 10.10.0.1:443: i/o timeout
```

Well okay I'm new to kubernetes may be there's something I didn't do right? Any help welcome.

**Comment by brianehlert** [comment]
It looks like you have missed the IngressClass. 
https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/#2-create-common-resources
Easy to miss when doing individual manifests.

As well as the CRDs:
https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/#create-custom-resources
`Note: By default, it is required to create custom resource definitions for VirtualServer, VirtualServerRoute, TransportServer and Policy. Otherwise, the Ingress Controller pods will not become Ready. If you’d like to disable that requirement, configure [-enable-custom-resources](https://docs.nginx.com/nginx-ingress-controller/configuration/global-configuration/command-line-arguments#cmdoption-global-configuration) command-line argument to false and skip this section.`

**Reply by brianehlert** [reply]
I recommend not skipping the CRDs.  They actually make lots of things easier.

**Comment by lucacome** [comment]
Hi @typoworx-de 

It looks like the Ingress Controller is not able to communicate with the API server. Could you double-check that networking, DNS, etc in the cluster are all working correctly?

Kubernetes docs have a few troubleshooting guides, one that I find particularly useful is this one https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution that shows how to create a test pod and also explains how to check that the DNS is working correctly.

**Comment by jjngx** [comment]
No response in 3 years, closing.
